stablehlo/reference/Ops.cpp

```diff
@@ -1074,6 +1093,28 @@ Tensor evalClzOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+Tensor evalCollectiveBroadcastOp(
+    const Tensor &operand, SmallVector<SmallVector<uint32_t>> replicaGroups,
+    ChannelId channelId, Process *process) {
+  if (!process)
+    llvm::report_fatal_error(
+        "collective_broadcast is only supported when run via "
+        "interpreter.run_parallel");
+
+  ProcessGroups processGroups;
+  if (channelId <= 0) processGroups = process->crossReplica(replicaGroups);
+  if (channelId > 0) processGroups = process->crossPartition(replicaGroups);
```

Comment:
Not blocking, outside the scope of this PR: I find this way of creating the process groups unintuitive. Why not use an enum to decide which communication scheme to use?

Also, when using `crossPartition` the `replicaGroups` are actually `partitionGroups`. Would `communicationGroups` be a better name?

Reply:
To answer the first question, it's written like that for spec parity since the spec uses them as function calls: 
https://openxla.org/stablehlo/spec#collective_broadcast
https://openxla.org/stablehlo/spec#collective_ops